doc/rationale.md

```diff
@@ -0,0 +1,33 @@
+# Ignition rationale
+
+Ignition is a distribution-agnostic, provisioning utility. When a machine boots for the first time, Ignition will manipulate it in various ways to set up the machine for operations.
+
+There are several principles that have guided Ignition's design. This document aims to explain these principles and how they impact Ignition's design.
+
+## Ignition only runs on the first boot
+
+Ignition is a provisioning tool; not a configuration management tool. Ignition definitely encourages immutable infrastructure, wherein if a user wishes to modify a machine, they should discard the old node and re-provision the machine. This keeps the user's machines in a well known state with relatively simple tooling.
+
+To this end, Ignition runs on the first boot of a machine. Being a general provisioning utility however, there's nothing stopping a user from using Ignition to set up some configuration management tool. This may even make a lot of sense depending on the organization and requirements the user is trying to meet.
+
+## Ignition produces the machine specified or no machine at all
+
+Ignition does what it needs to make the system match the state described in the Ignition config. If for any reason Ignition cannot deliver the exact machine that the config asked for, Ignition prevents the machine from booting successfully.
+
+For example, if the user wanted to fetch the document hosted at `https://example.com/foo.conf` and write it to disk, Ignition would prevent the machine from booting in the event that it is unable to resolve the given URL.
+
+## Ignition configs are declarative
+
+Ignition configs describe the state of a system. Ignition configs do _not_ list a series of steps that Ignition should take.
+
+To this end, Ignition configs do not allow users to provide arbitrary logic (including scripts for Ignition to run). Users describe which filesystems they want to exist, what files need to be created, what users should exist, and more. Any further customization needs to be using systemd services, created by Ignition.
+
+## Ignition configs should not be written by hand
+
+Ignition configs are not intended to be human friendly and users should not write Ignition configs by hand. Ignition configs were designed to be human readable, but difficult to write. This was done to discourage users from attempting to write configs by hand.
+
+Instead, a separate tool should be used to generate Ignition configs. In the case of Container Linux, files called Container Linux Configs can be written in YAML and then converted into Ignition configs. This conversion process allows for distribution specific configs to be converted into a distribution agnostic Ignition config, and ensures that the Ignition config gets validated before a user attempts to use it.
+
+## Ignition runs in the initramfs
```

Comment:
I agree with @crawford; this is an implementation detail, not part of the rational. Drop this section. You could replace it with something like "Ignition should configure the boot process before beginning it" (please find better wording than that). Or maybe combine it with "Ignition runs only on first boot" section to mention that the first boot is not "special". i.e. your system is ready to go on the first boot and there's no extra reboot needed. (again, that wording is bad)